persistence/modeshape-persistence-relational/src/main/resources/org/modeshape/persistence/relational/sqlserver_database.properties

```diff
@@ -30,4 +30,7 @@ update_content = UPDATE {0} SET LAST_CHANGED=CURRENT_TIMESTAMP, CONTENT = ? WHER
 remove_content = DELETE FROM {0} WHERE ID IN (#)
 
 # Remove all documents
-remove_all_content = DELETE FROM {0}
\ No newline at end of file
+remove_all_content = DELETE FROM {0}
+
+# Lock documents
+lock_content = SELECT ID FROM {0} WHERE ID IN (#) WITH (UPDLOCK) 
```

Comment:
Should `ROWLOCK` hint also be used here in addition to `UPDLOCK`? From the [documentation on SQL Server hints](https://msdn.microsoft.com/en-us/library/ms187373.aspx):

> UPDLOCK
> Specifies that update locks are to be taken and held until the transaction completes. UPDLOCK takes update locks for read operations only at the row-level or page-level....

and

> ROWLOCK
> Specifies that row locks are taken when page or table locks are ordinarily taken. When specified in transactions operating at the SNAPSHOT isolation level, row locks are not taken unless ROWLOCK is combined with other table hints that require locks, such as UPDLOCK and HOLDBACK.

So if the database is using SNAPSHOT isolation level, then an `UPDLOCK` hint doesn't do anything.

> SNAPSHOT
> ... The transaction can only recognize data modifications that were committed before the start of the transaction. Data modifications made by other transactions after the start of the current transaction are not visible to statements executing in the current transaction. The effect is as if the statements in a transaction get a snapshot of the committed data as it existed at the start of the transaction.

So if SNAPSHOT isolation were used with MySQL, then using just `UPDLOCK` means that no locking actually happens, and coupled with SNAPSHOT isolation means that two competing transactions might both read a node's state, the second txn to commit an update will overwrite the first txn's update and the first transaction's changes *will be lost*. However, using both the `UPDLOCK,ROWLOCK` hints would cause the database to acquire locks even when using SNAPSHOT isolation, meaning the transactions get serialized.